### ISSUE TITLE
update(n=0) causes ZeroDivisionError if it is the first update

### ISSUE THREAD

**Issue by ferrouswheel** [tqdm/tqdm#268]
v4.8.4

```
Traceback (most recent call last):
  File "x.py", line 273, in train
    bar.update(0)
  File "C:\Python27\lib\site-packages\tqdm\_tqdm.py", line 808, in update
    if self.avg_time is None \
ZeroDivisionError: float division by zero
```

If one is manually updating a progress bar, and for some reason your code can output a step size of 0 (while smoothing the time estimate), then a ZeroDivisionError occurs.

This only happens if `update(n=0)` is the first call. If you have previously called `update` with a non-zero number, then subsequent updates with `n==0` are fine.

**Comment by CrazyPython** [comment]
`train` yo bash script, boi?

(sorry in advance for that tone)

**Comment by ferrouswheel** [comment]
Yup - run from terminal.

Seems to only happen if some time has passed between instantiation of tqdm and the update call. e.g.

```python
from tqdm import tqdm, trange
bar = tqdm(total=10)
bar.update(0)
```

is fine... but this crashes:

```python
import time
from tqdm import tqdm, trange
bar = tqdm(total=10)
time.sleep(1)
bar.update(0)
```

**Comment by CrazyPython** [comment]
@ferrouswheel very weird. somebody boot up the debugger.

**Comment by lrq3000** [comment]
This can be reproduced like this without delay:

```
import time
from tqdm import tqdm, trange
bar = tqdm(total=10, mininterval=0)
time.sleep(0.001)
bar.update(0)
```

The bug came from this line: https://github.com/tqdm/tqdm/blob/master/tqdm/_tqdm.py#L806

Because it didn't check if delta_it > 0.

This will be fixed, thank's for the nice catch and clean report @ferrouswheel, this was a hard bug to spot!

**Comment by lrq3000** [comment]
Closed by #264.
